### PR TITLE
Libsigcpp conanv2

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **libsigcpp**

Conan v2 and new toolchain updates

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by planetmarshall** [comment]
The `MesonToolchain` does not appear to be passing the `-stdlib=libc++` option when building with clang. I don't see any other C++ projects using the new `MesonToolchain`

**Comment by uilianries** [comment]
@planetmarshall There are some projects using MesonToolchain:

https://github.com/conan-io/conan-center-index/blob/a31f74c666182039eb9a64eca076ff881898dd11/recipes/inih/all/conanfile.py

https://github.com/conan-io/conan-center-index/blob/e42d954b4e68d03e7543bf22f4b95ce10b24e20a/recipes/serd/all/conanfile.py

https://github.com/conan-io/conan-center-index/blob/bc6cb0a6eff0864304f1029ee9cc7fcb17b1fe2d/recipes/dav1d/all/conanfile.py

https://github.com/conan-io/conan-center-index/blob/ad68e025eeb05270771dfd2bc8da2f4c0a133eca/recipes/libglvnd/all/conanfile.py

https://github.com/conan-io/conan-center-index/blob/95482a148d2f48d4fd6a0465b28dd38a32c25e1d/recipes/libsystemd/all/conanfile.py

https://github.com/conan-io/conan-center-index/blob/79388bda299cab5da326027d6447e29284f38f0f/recipes/wayland/all/conanfile.py

**Comment by planetmarshall** [comment]
> @planetmarshall There are some projects using MesonToolchain:
> 

Yes, these are all C projects though. They all delete the `libcxx` setting.

**Comment by uilianries** [comment]
@planetmarshall In that case, it could be a generator limitation. It may worth opening an issue to Conan client asking about it.

**Comment by planetmarshall** [comment]
> @planetmarshall In that case, it could be a generator limitation. It may worth opening an issue to Conan client asking about it.

See conan-io/conan#12009

File: recipes/libsigcpp/2.x.x/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,8 @@
 cmake_minimum_required(VERSION 3.6)
 project(test_package)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGET)
-
-find_package(PkgConfig REQUIRED)
-pkg_search_module(SIGCPP IMPORTED_TARGET REQUIRED sigc++-2.0)
+find_package(libsigcpp REQUIRED)
```
Comment (planetmarshall) [inline recipes/libsigcpp/2.x.x/test_package/CMakeLists.txt:5]: This doesn't appear to work with the new `PkgConfigDeps` generator - not sure if the recipe is missing something or there's an existing issue.

Comment (uilianries) [reply recipes/libsigcpp/2.x.x/test_package/CMakeLists.txt:5]: You touched a known bug: https://github.com/conan-io/conan/issues/11962

Basically you need a workaround to use pkg_search_module with new cmake generators: https://github.com/conan-io/conan-center-index/blob/master/recipes/libcap/all/test_package/conanfile.py

Not beautiful, but should work.

**Comment by planetmarshall** [comment]
Duplicate pf #14778

**Pull request closed without merging**